### PR TITLE
feat: pass custom credentials as additional arguments of insights methods (v1)

### PR DESCRIPTION
**Pull request by dhayab**
See https://github.com/algolia/search-insights.js/pull/398

### CODE REVIEW

File: lib/__tests__/click.test.ts [hunk 1]
```diff
@@ -56,86 +65,117 @@ describe("clickedObjectIDsAfterSearch", () => {
...
+      undefined
+    );
+  });
 
+  test("should call sendEvents with additional params if provided", () => {
+    analyticsInstance.clickedObjectIDsAfterSearch(
+      clickParams,
+      additionalParameters
+    );
     expect((analyticsInstance as any).sendEvent).toHaveBeenCalledWith(
```
Comment (Haroenv) [inline lib/__tests__/click.test.ts:89]: I guess it's consistent, but do you know why they have to be casted?

File: lib/_sendEvent.ts [hunk 2]
```diff
@@ -143,9 +146,21 @@ function bulkSendEvent(
   apiKey: string,
   userAgent: string,
   endpointOrigin: string,
-  events: InsightsEvent[]
+  events: InsightsEvent[],
+  additionalHeaders: InsightsAdditionalEventParams["headers"] = {}
```
Comment (Haroenv) [inline lib/_sendEvent.ts:150]: technically query params, but I guess it's equivalent

**Review by Haroenv — APPROVED** [review]

File: lib/__tests__/click.test.ts (hunk 1 above)
Comment (dhayab) [reply lib/__tests__/click.test.ts:89]: In v1 `AlgoliaAnalytics.sendEvent()` is a protected method. I could replace this with `// @ts-expect-error` as I did for other assertions in those tests if you prefer.

File: lib/_sendEvent.ts (hunk 2 above)
Comment (dhayab) [reply lib/_sendEvent.ts:150]: Yes I wondered but I thought that although sent as query parameters, they are first custom headers.

File: lib/__tests__/click.test.ts (hunk 1 above)
Comment (Haroenv) [reply lib/__tests__/click.test.ts:89]: either way is fine I think

**Review by sarahdayan — APPROVED** [review]
Let's also update the README.

**Pull request merged**